nibabel/cifti2/cifti2.py

```diff
@@ -0,0 +1,1115 @@
+            mat_ind_map.append(map_._to_xml_element())
+
+        return mat_ind_map
+
+
+class Cifti2Matrix(xml.XmlSerializable, collections.MutableSequence):
+    def __init__(self):
+        self._mims = []
+        self.metadata = None
+
+    @property
+    def metadata(self):
+        return self._meta
+
+    @metadata.setter
+    def metadata(self, meta):
+        """ Set the metadata for this Cifti2Header
+
+        Parameters
+        ----------
+        meta : Cifti2MetaData
+
+        Returns
+        -------
+        None
+        """
+        self._meta = _value_if_klass(meta, Cifti2MetaData, True)
+
+    def __setitem__(self, key, value):
+        if not isinstance(value, Cifti2MatrixIndicesMap):
+            raise TypeError("Not a valid Cifti2MatrixIndicesMap instance")
+        self._mims[key] = value
+
+    def __getitem__(self, key):
+        return self._mims[key]
+
+    def __delitem__(self, key):
+        del self._mims[key]
+
+    def __len__(self):
+        return len(self._mims)
+
+    def insert(self, index, value):
+        if not isinstance(value, Cifti2MatrixIndicesMap):
+            raise TypeError("Not a valid Cifti2MatrixIndicesMap instance")
+        self._mims.insert(index, value)
+
+    def _to_xml_element(self):
+        if (len(self) == 0 and self.metadata is None):
+            raise CIFTI2HeaderError(
+                'Matrix element requires either a MatrixIndicesMap or a Metadata element'
+            )
+
+        mat = xml.Element('Matrix')
+        if self.metadata:
+            mat.append(self.metadata._to_xml_element())
+        for mim in self._mims:
+            mat.append(mim._to_xml_element())
+        return mat
+
+
+class Cifti2Header(FileBasedHeader, xml.XmlSerializable):
+    ''' Class for Cifti2 header extension '''
+
+    def __init__(self, matrix=None, version="2.0"):
+        FileBasedHeader.__init__(self)
+        xml.XmlSerializable.__init__(self)
+        self.matrix = Cifti2Matrix() if matrix is None else Cifti2Matrix()
+        self.version = version
+
+    def _to_xml_element(self):
+        cifti = xml.Element('CIFTI')
+        cifti.attrib['Version'] = str(self.version)
+        mat_xml = self.matrix._to_xml_element()
+        if mat_xml is not None:
+            cifti.append(mat_xml)
+        return cifti
+
+    @classmethod
+    def from_header(klass, header=None):
+        if header is None:
+            return klass()
+        if type(header) == klass:
+            return header.copy()
+        raise ValueError('header is not a Cifti2Header')
+
+    @classmethod
+    def may_contain_header(klass, binaryblock):
+        from .parse_cifti2_fast import _Cifti2AsNiftiHeader
+        return _Cifti2AsNiftiHeader.may_contain_header(binaryblock)
+
+
+class Cifti2Image(FileBasedImage):
+    """ Class for single file CIfTI2 format image
+    """
+    header_class = Cifti2Header
+    valid_exts = Nifti2Image.valid_exts
+    files_types = Nifti2Image.files_types
+    makeable = False
```

Comment:
Pretty sure it was just because that was a safe default when Ben split out `FileBasedImage` from `SpatialImage`, but looking through the tests, we only check `makeable` on `SpatialImage`s, anyway.